piccolo/query/methods/insert.py

```diff
@@ -15,14 +20,24 @@
 class Insert(
     t.Generic[TableInstance], Query[TableInstance, t.List[t.Dict[str, t.Any]]]
 ):
-    __slots__ = ("add_delegate", "returning_delegate")
+    __slots__ = (
+        "add_delegate",
+        "returning_delegate",
+        "on_conflict_delegate",
+    )
 
     def __init__(
-        self, table: t.Type[TableInstance], *instances: TableInstance, **kwargs
+        self,
+        table: t.Type[TableInstance],
+        on_conflict: t.Optional[OnConflict] = None,
```

Comment:
I would drop consider dropping this, and just using the `on_conflict` method instead.

The reason being, we can add extra arguments to the `on_conflict` method in the future, but we don't want to add too many to `__init__`.

Reply:
I don't think we should change it, but feel free to do as you think is best. With this method, we can also easily solve the [problem with duplicate entries in M2M](https://github.com/piccolo-orm/piccolo/issues/707). I'm sorry if I didn't understand well what you wanted to say and feel free to change this if you think `on_conflict` as method would be better way.